mqlc/builtin_array.go

```diff
@@ -53,7 +53,7 @@ func compileWhere(c *compiler, typ types.Type, ref uint64, id string, call *pars
 		chunkId := "==" + string(childType)
 		if blockTyp != childType {
 			chunkId = "==" + string(blockTyp)
-			_, err := llx.BuiltinFunctionV2(blockTyp, chunkId)
+			_, err := llx.BuiltinFunctionV2(childType, chunkId)
```

Comment:
^^ additionally this was always wrong. It just inserted the operation `T == T`, which doesn't make any sense